### PR TITLE
update slices

### PR DESCRIPTION
**Pull request by wilzbach**

### CODE REVIEW

File: public/content/en/basics.md [hunk 1]
```diff
@@ -76,27 +76,39 @@ If no other value is given in the declaration all integers
 are initialized with `0` and floating points
 with `NaN` (*not a number*).
 
+In D indexes have usually the alias type `size_t` as it a type that
```
Comment (stonemaster) [inline public/content/en/basics.md]: "is a type"

File: public/content/en/basics.md [hunk 2]
```diff
@@ -76,27 +76,39 @@ If no other value is given in the declaration all integers
...
+is large enough to represent an offset into all addressible memory - it is
+be `uint` for 32bit and `ulong` for 64bit architectures.
```
Comment (stonemaster) [inline public/content/en/basics.md]: "that is `uint` for 32-bit and `along` for 64-bit"

File: public/content/en/basics.md [hunk 3]
```diff
@@ -76,27 +76,39 @@ If no other value is given in the declaration all integers
...
+
+`assert` is compiler built-in which verifies conditions in debug mode and aborts
+with an AssertionError if it fails.
```
Comment (stonemaster) [inline public/content/en/basics.md]: `AssertionError`

File: public/content/en/basics.md [hunk 4]
```diff
@@ -76,27 +76,39 @@ If no other value is given in the declaration all integers
...
 ## {SourceCode}
 
 import std.stdio;
 
 void main()
 {
-    // Big numbers can be made more
-    // readable with an "_"; it's still
-    // a 7000000 though.
+    // Big number can be seperated
```
Comment (stonemaster) [inline public/content/en/basics.md]: numbers

File: public/content/en/basics.md [hunk 5]
```diff
@@ -76,27 +76,39 @@ If no other value is given in the declaration all integers
...
+    // with a underscore "_"
```
Comment (stonemaster) [inline public/content/en/basics.md]: with an

File: public/content/en/basics.md [hunk 6]
```diff
@@ -76,27 +76,39 @@ If no other value is given in the declaration all integers
...
     auto f = 3.1415f; // .f denotes a float
+
     // typeid(VAR) returns the type information
     // of an expression.
     writeln("type of f is ", typeid(f));
     double pi = f; // fine
-    // would be an error:
-    // float bad = pi;
+
+    // implicit down-casting yields an error
```
Comment (stonemaster) [inline public/content/en/basics.md]: Make full stop after comment.

File: public/content/en/basics.md [hunk 7]
```diff
@@ -76,27 +76,39 @@ If no other value is given in the declaration all integers
...
+    // with traits and assert we can check whether code compiles
+    // it will explained in deep soon
```
Comment (stonemaster) [inline public/content/en/basics.md]: Would remove line

File: public/content/en/basics.md [hunk 8]
```diff
@@ -76,27 +76,39 @@ If no other value is given in the declaration all integers
...
+
     // typeid(VAR) returns the type information
     // of an expression.
     writeln("type of f is ", typeid(f));
     double pi = f; // fine
-    // would be an error:
-    // float bad = pi;
+
+    // implicit down-casting yields an error
+    // with traits and assert we can check whether code compiles
```
Comment (stonemaster) [inline public/content/en/basics.md]: "The following static assert checks whether code compiles:"

**Comment by wilzbach** [comment]
rebased :)

File: public/content/en/basics.md [hunk 9]
```diff
@@ -76,26 +76,37 @@ If no other value is given in the declaration all integers
 are initialized with `0` and floating points
 with `NaN` (*not a number*).
 
+In D indexes have usually the alias type `size_t` as it is a type that
+is large enough to represent an offset into all addressible memory - that is
+`uint` for 32bit and `ulong` for 64bit architectures.
```
Comment (stonemaster) [inline public/content/en/basics.md]: 32-bit and 64-bit

File: public/content/en/basics.md [hunk 10]
```diff
@@ -76,26 +76,37 @@ If no other value is given in the declaration all integers
...
 ## {SourceCode}
 
 import std.stdio;
 
 void main()
 {
-    // Big numbers can be made more
-    // readable with an "_"; it's still
-    // a 7000000 though.
+    // Big numbers can be seperated
```
Comment (stonemaster) [inline public/content/en/basics.md]: separated

**Comment by wilzbach** [comment]
rebased

**Pull request merged**